File: src/types/wecom.ts
// 企微微盘扫描 API 的 DTO 类型（Path A）。
// 响应只含安全运营元数据：绝不含微盘 file_id / 下载 URL / access_token / cookie /
// 存储引用 / WeKnora id / 业务原文。

export interface WecomScanConfigDTO {
  id: string;
  name: string | null;
  scope_type: string;
  related_project_id: string | null;
  related_project_name: string | null;
  scan_space_status: string;
  manager_access_status: string;
  enabled: boolean;
  // created_by = 待确认任务业务归属人（非配置操作人 admin）。
  created_by: string;
  task_owner_name: string | null;
  task_owner_role_label: string | null;
  scan_frequency: string | null;
  last_scan_at: string | null;
  created_at: string;
  updated_at: string;
}

export interface WecomScanConfigsResponseDTO {
  items: WecomScanConfigDTO[];
}

// 创建项目扫描配置。扫描空间和根目录只由后端创建、保存和解析。
export interface WecomScanConfigCreateBody {
  name: string;
  target_project_id: string;
  task_owner_user_id: string;
  enabled: boolean;
}

// 编辑请求（全部可选，仅传入字段更新；含启停 / 改归属人）。
export interface WecomScanConfigUpdateBody {
  name?: string;
  task_owner_user_id?: string;
  enabled?: boolean;
}

export interface WecomProjectOptionDTO {
  id: string;
  name: string;
  scan_space_status: string;
  manager_access_status: string;
}

export interface WecomProjectOptionsResponseDTO {
  items: WecomProjectOptionDTO[];
}

// 业务归属人候选（仅安全字段）。project_ids / is_governance 供前端按 scope 提示。
export interface WecomOwnerOptionDTO {
  user_id: string;
  name: string;
  role_label: string | null;
  project_ids: string[];
  is_governance: boolean;
}

export interface WecomOwnerOptionsResponseDTO {
  items: WecomOwnerOptionDTO[];
}

export interface WecomScanRecordDTO {
  id: string;
  config_id: string;
  trace_id: string | null;
  scan_started_at: string;
  scan_completed_at: string | null;
  discovered_count: number;
  new_count: number;
  duplicate_count: number;
  failed_count: number;
  scan_status: string;
  error_type: string | null;
  error_message: string | null;
  created_at: string;
}

export interface WecomScanRecordsResponseDTO {
  items: WecomScanRecordDTO[];
}

export interface WecomAuthorizeDTO {
  authorize_url: string;
}

File: backend/app/schemas/wecom.py
"""企微微盘扫描 API 的请求 / 响应 schema（Path A）。

响应只含安全运营元数据：**绝不**含内部存储引用 / 源文件引用 / 微盘下载 URL /
微盘 file_id / access_token / WeKnora id / 业务原文。
"""

from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel


class WecomScanConfigOut(BaseModel):
    id: uuid.UUID
    name: str | None
    scope_type: str
    related_project_id: uuid.UUID | None
    related_project_name: str | None = None
    scan_space_status: str
    manager_access_status: str
    enabled: bool
    # created_by 即"待确认任务业务归属人"：扫描产物的 IngestTask.created_by。
    # 配置操作人是当前 admin（见审计 actor），与此业务归属人不是同一概念。
    created_by: uuid.UUID
    task_owner_name: str | None = None
    task_owner_role_label: str | None = None
    scan_frequency: str | None
    last_scan_at: datetime | None
    created_at: datetime
    updated_at: datetime


class WecomScanConfigsResponse(BaseModel):
    items: list[WecomScanConfigOut]


class WecomScanConfigCreateBody(BaseModel):
    """创建项目扫描配置；扫描根固定为服务端项目扫描空间根目录。"""

    name: str
    target_project_id: uuid.UUID
    task_owner_user_id: uuid.UUID
    enabled: bool = True


class WecomScanConfigUpdateBody(BaseModel):
    """编辑项目扫描配置；项目和扫描空间不可由浏览器修改。"""

    name: str | None = None
    task_owner_user_id: uuid.UUID | None = None
    enabled: bool | None = None


class WecomProjectOptionOut(BaseModel):
    """项目候选项（仅安全字段：id + 名称），供创建/编辑配置时选择目标项目。"""

    id: uuid.UUID
    name: str
    scan_space_status: str
    manager_access_status: str


class WecomProjectOptionsResponse(BaseModel):
    items: list[WecomProjectOptionOut]


class WecomOwnerOptionOut(BaseModel):
    """业务归属人候选（仅安全字段）：active 业务用户。

    project_ids / is_governance 供前端按 target_scope 做候选提示；后端最终校验为准。
    绝不含 token / session / wecom_user_id 明文 / ip / device。
    """

    user_id: uuid.UUID
    name: str
    role_label: str | None = None
    project_ids: list[uuid.UUID] = []
    is_governance: bool = False


class WecomOwnerOptionsResponse(BaseModel):
    items: list[WecomOwnerOptionOut]


class WecomScanRecordOut(BaseModel):
    id: uuid.UUID
    config_id: uuid.UUID
    trace_id: str | None
    scan_started_at: datetime
    scan_completed_at: datetime | None
    discovered_count: int
    new_count: int
    duplicate_count: int
    failed_count: int
    scan_status: str
    error_type: str | None
    error_message: str | None
    created_at: datetime


class WecomScanRecordsResponse(BaseModel):
    items: list[WecomScanRecordOut]
